components/OAuth2.0/authorization.c

```diff
@@ -0,0 +1,282 @@
+            {
+                ESP_LOGW(TAG, "token is expired");
+                bool IsTokenRenewed = Oauth_TokenRenew();
+                if (!IsTokenRenewed)                                                                   // Run function to renew the token
+                {
+                    AuthInterfaceHandler->Status = LOGIN;
+                    break;                                                                   // set Status to ILDLE if token renewed unsuccessfully
+                }
+
+                xSemaphoreGive((*AuthInterfaceHandler->IsServiceAuthorizedSemaphore));                              // give IsServiceAuthorizedSemaphore semaphore in the IntefaceHandler
+                AuthInterfaceHandler->TokenLastUpdate = xTaskGetTickCount();                                           // Save the time when the token was received
+                AuthInterfaceHandler->Status = CHECK_TIME;                                                             // set Status to CHECK_TIME if token renewed successfully
+                break;
+            }
+        }   
+        vTaskDelay(pdMS_TO_TICKS(100)); //FIXME magic number                                                                            // Task delay at the end of while(1) loop
+    }
+}
+
+/**
+ * @brief This function initiates the authorization process.
+ * @param InterfaceHandler as the handler
+ * @return true if task run to the end
+ */
+bool Oauth_TaskInit(OAuthInterfaceHandler_t *InterfaceHandler)
+{
+    AuthInterfaceHandler = InterfaceHandler;
+    AuthInterfaceHandler->Status = INIT;
+    if (AuthInterfaceHandler->ConfigAddressInSpiffs != NULL &&
+        AuthInterfaceHandler->IsServiceAuthorizedSemaphore != NULL)
+    {
+        StaticTask_t *xTaskBuffer = (StaticTask_t *)malloc(sizeof(StaticTask_t));
+        StackType_t *xStack = 
+                (StackType_t *)malloc(SERVICE_TASK_STACK_SIZE * sizeof(StackType_t));                 // Assuming a stack size of 400 words (adjust as needed)
+        if (xTaskBuffer == NULL || xStack == NULL)
+        {
+            ESP_LOGE(TAG, "Memory allocation failed!\n");
+            free(xTaskBuffer);
+            free(xStack);
+            return false; // Exit with an error code
+        }
+        xTaskCreateStatic(
+            Oauth_MainTask,                    // Task function
+            "Oauth_MainTask",                  // Task name (for debugging)
+            SERVICE_TASK_STACK_SIZE,             // Stack size (in words)
+            NULL,                                // Task parameters (passed to the task function)
+            tskIDLE_PRIORITY + SERVICE_PRIORITY, // Task priority (adjust as needed)
+            xStack,                              // Stack buffer
+            xTaskBuffer                          // Task control block
+        );
+
+        // Allocate buffer and Initialize the service API call //TODO change dynamic mem to static
+        AuthInterfaceHandler->OAuthBuffer.MessageBuffer =
```

Comment:
same comment. this buffer is private, should not be exposed

Reply:
fixed.